### PR TITLE
refactor!: move SSS coordinate information to spacePointData

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
This PR moves the calculation of `topStripVector` and `bottomStripVector` to `spacePointData`, so we can calculate these values only once per event, since these values are calculated for ~95% of the input space points anyway.

@CarloVarni

### CODE REVIEW

File: Core/include/Acts/Seeding/SeedFinderUtils.ipp [hunk 1]
```diff
@@ -178,56 +178,61 @@ inline bool xyzCoordinateCheck(
...
+  //  spacePointData.getTopHalfStripLength(index); const float&
+  //  bottomHalfStripLength =
+  //      spacePointData.getBottomHalfStripLength(index);
+  //  const Acts::Vector3& topStripDirection =
+  //      spacePointData.getTopStripDirection(index);
+  //  const Acts::Vector3& bottomStripDirection =
+  //      spacePointData.getBottomStripDirection(index);
+
+  // prepare variables
+  const Acts::Vector3& topStripVector = spacePointData.getTopStripVector(index);
```
Comment (CarloVarni) [inline Core/include/Acts/Seeding/SeedFinderUtils.ipp:182]: `Acts::Vector3` uses `double`. So we may want to remain to `double` here and in the other places. We can then evaluate what is the effect of moving from `double` to `float` but I'd keep that for another PR

File: Core/include/Acts/Seeding/SeedFinderUtils.ipp [hunk 2]
```diff
@@ -178,56 +178,61 @@ inline bool xyzCoordinateCheck(
...
-  double zTopStripVector = topHalfStripLength * topStripDirection[2];
-  double xBottomStripVector = bottomHalfStripLength * bottomStripDirection[0];
-  double yBottomStripVector = bottomHalfStripLength * bottomStripDirection[1];
-  double zBottomStripVector = bottomHalfStripLength * bottomStripDirection[2];
+  const float& xTopStripVector = topStripVector[0];
+  const float& yTopStripVector = topStripVector[1];
+  const float& zTopStripVector = topStripVector[2];
+  const float& xBottomStripVector = bottomStripVector[0];
+  const float& yBottomStripVector = bottomStripVector[1];
+  const float& zBottomStripVector = bottomStripVector[2];
```
Comment (CarloVarni) [inline Core/include/Acts/Seeding/SeedFinderUtils.ipp:193]: these are `double`

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp [hunk 3]
```diff
@@ -259,14 +259,46 @@ ActsExamples::ProcessCode ActsExamples::SeedingAlgorithm::execute(
...
-        state.spacePointData.setTopStripDirection(
-            index, m_cfg.seedFinderConfig.getTopStripDirection(sp->sp()));
-        state.spacePointData.setBottomStripDirection(
-            index, m_cfg.seedFinderConfig.getBottomStripDirection(sp->sp()));
+
+        const float& topHalfStripLength =
+            m_cfg.seedFinderConfig.getTopHalfStripLength(sp->sp());
+        const float& bottomHalfStripLength =
+            m_cfg.seedFinderConfig.getBottomHalfStripLength(sp->sp());
+        const Acts::Vector3& topStripDirection =
```
Comment (CarloVarni) [inline Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp:267]: all these function return objects by value, you cannot use `const Acts::Vector3&`

**Comment by CarloVarni** [comment]
Do you know if this has an important impact on the timings?

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp (hunk 3 above)
Comment (LuisFelipeCoelho) [reply Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp:267]: I am confused on this point, don't getTopStripDirection and getBottomStripDirection return a Vector3?

**Comment by LuisFelipeCoelho** [comment]
This improves SSS by ~12% , doesn't affect PPP

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp (hunk 3 above)
Comment (CarloVarni) [reply Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp:267]: I'm suggesting to change this to the following:
```cpp
        const float topHalfStripLength =
            m_cfg.seedFinderConfig.getTopHalfStripLength(sp->sp());
        const float bottomHalfStripLength =
            m_cfg.seedFinderConfig.getBottomHalfStripLength(sp->sp());
        const Acts::Vector3 topStripDirection =
            m_cfg.seedFinderConfig.getTopStripDirection(sp->sp());
        const Acts::Vector3 bottomStripDirection =
            m_cfg.seedFinderConfig.getBottomStripDirection(sp->sp());

        state.spacePointData.setTopStripVector(
            index, topHalfStripLength * topStripDirection);
        state.spacePointData.setBottomStripVector(
            index,
            bottomHalfStripLength * bottomStripDirection);
        state.spacePointData.setStripCenterDistance(
            index, m_cfg.seedFinderConfig.getStripCenterDistance(sp->sp()));
        state.spacePointData.setTopStripCenterPosition(
            index, m_cfg.seedFinderConfig.getTopStripCenterPosition(sp->sp()));
```

Comment (LuisFelipeCoelho) [reply Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp]: ah of course, sorry

**Review by CarloVarni — APPROVED** [review]
